sharding-core/sharding-core-execute/src/main/java/org/apache/shardingsphere/core/execute/metadata/TableMetaDataLoader.java

```diff
@@ -103,60 +107,66 @@ public TableMetaData load(final String logicTableName, final ShardingRule shardi
             }
         });
     }
-    
-    private Collection<TableMetaData> load(final String dataSourceName, final String catalog, 
-                                           final String logicTableName, final Collection<DataNode> dataNodes, final String generateKeyColumnName, final EncryptRule encryptRule) throws SQLException {
+
+    private Collection<TableMetaData> load(final String dataSourceName, final String catalog,
+            final String logicTableName, final Collection<DataNode> dataNodes, final String generateKeyColumnName, final EncryptRule encryptRule) throws SQLException {
         Collection<TableMetaData> result = new LinkedList<>();
         try (Connection connection = connectionManager.getConnection(dataSourceName)) {
+            DatabaseMetaData metaData = connection.getMetaData();
+            String databaseName = metaData.getDatabaseProductName();
+            String schema = null;
+            if (databaseName != null && "oracle".equalsIgnoreCase(databaseName)) {
```

Comment:
Catalog and schemaName are different, so we should store them togather in `DatabaseMetaData` ahead. Consequently, you can get catalog or schema from `DatabaseMetaData` when need. Add an interface named `getCatalog()` for `DataSourceMetaData` is necessary.